desc/grid.py

```diff
@@ -283,7 +295,7 @@ def unique_rho_idx(self):
         errorif(
             not hasattr(self, "_unique_rho_idx"),
             AttributeError,
-            f"{self} does not have unique indices assigned. "
+            "Grid does not have unique indices assigned. "
```

Comment:
I kind of prefer having it print `self` in case there are multiple grids so its clear which one its referring to

Reply:
The issue is that calls the print / to string method of the grid object, which calls the L, M, N attributes, which calls this method again, and you get stuck recursing.